src/System.Private.ServiceModel/src/System/IdentityModel/CryptoHelper.cs

```diff
@@ -61,7 +97,305 @@ internal static bool IsAsymmetricAlgorithm(string algorithm)
+            }
+            if (algorithmObject != null)
+            {
+                SymmetricAlgorithm symmetricAlgorithm = algorithmObject as SymmetricAlgorithm;
+                KeyedHashAlgorithm keyedHashAlgorithm = algorithmObject as KeyedHashAlgorithm;
+
+                if (symmetricAlgorithm != null || keyedHashAlgorithm != null)
+                    found = true;
+                // The reason we do not return here even when the user has provided a custom algorithm to CryptoConfig 
+                // is because we need to check if the user has overwritten an existing standard URI.
+            }
+
+            switch (algorithm)
+            {
+                case SecurityAlgorithms.DsaSha1Signature:
+                case SecurityAlgorithms.RsaSha1Signature:
+                case SecurityAlgorithms.RsaSha256Signature:
+                case SecurityAlgorithms.RsaOaepKeyWrap:
+                case SecurityAlgorithms.RsaV15KeyWrap:
+                    return false;
+                case SecurityAlgorithms.HmacSha1Signature:
+                case SecurityAlgorithms.HmacSha256Signature:
+                case SecurityAlgorithms.Psha1KeyDerivation:
+                case SecurityAlgorithms.Psha1KeyDerivationDec2005:
+                    return true;
+                case SecurityAlgorithms.Aes128Encryption:
+                case SecurityAlgorithms.Aes128KeyWrap:
+                    return keySize >= 128 && keySize <= 256;
+                case SecurityAlgorithms.Aes192Encryption:
+                case SecurityAlgorithms.Aes192KeyWrap:
+                    return keySize >= 192 && keySize <= 256;
+                case SecurityAlgorithms.Aes256Encryption:
+                case SecurityAlgorithms.Aes256KeyWrap:
+                    return keySize == 256;
+                case SecurityAlgorithms.TripleDesEncryption:
+                case SecurityAlgorithms.TripleDesKeyWrap:
+                    return keySize == 128 || keySize == 192;
+                default:
+                    if (found)
+                        return true;
+                    return false;
+                    // We do not expect the user to map the uri of an existing standrad algorithm with say key size 128 bit 
+                    // to a custom algorithm with keySize 192 bits. If he does that, we anyways make sure that we return false.
+            }
+        }
+
+        internal static void FillRandomBytes(byte[] buffer)
+        {
+            RandomNumberGenerator.GetBytes(buffer);
+        }
+
+        /// <summary>
+        /// This generates the entropy using random number. This is usually used on the sending 
+        /// side to generate the requestor's entropy.
+        /// </summary>
+        /// <param name="data">The array to fill with cryptographically strong random nonzero bytes.</param>
+        public static void GenerateRandomBytes(byte[] data)
+        {
+            RandomNumberGenerator.GetNonZeroBytes(data);
```

Comment:
What target are you compiling for?  If it's netcoreapp2.1 (or higher) or netstandard2.1 then you can use the new static method `RandomNumberGenerator.Fill(data)`.  (Assuming you don't need the "non-zero" part)

Reply:
That method doesn't look to provide any benefit over GetBytes. The benefit looks to be creating a Span for a part of an array and passing the Span to do a partial fill as that will be more efficient than GetBytes(byte[] data, int offset, int count) which does a copy. But when filling the entire array, this seems equivalent. But nice to know cleaner api's are available.

---
In reply to: [273996432](https://github.com/dotnet/wcf/pull/3538#discussion_r273996432) [](ancestors = 273996432)

Reply:
Well, `Fill` is static, so you don't have to decide how you want to manage the object described as not thread-safe :smile:

Reply:
I just tried to change it and realized we're targeting netstandard2.0 still. I checked the implementation of what's returned by RandomNumberGenerator.Create() and the Fill method and they do the same thing, they all eventually call the same static method in RandomNumberGeneratorImplementation.

---
In reply to: [274725326](https://github.com/dotnet/wcf/pull/3538#discussion_r274725326) [](ancestors = 274725326)